python/GafferOSLTest/OSLObjectTest.py

```diff
@@ -236,5 +236,137 @@ def testReferencePromotedPlug( self ) :
 
 		s["r"]["p"].setInput( s["s"]["out"] )
 
+	def testCanShadeVertexInterpolatedPrimitiveVariablesAsUniform( self ) :
+
+		s = Gaffer.ScriptNode()
+
+		c = GafferScene.Cube()
+		s.addChild( c )
+
+		o = GafferOSL.OSLObject()
+		s.addChild( o )
+
+		f = GafferScene.PathFilter( "PathFilter" )
+		s.addChild( f )
+		f["paths"].setValue( IECore.StringVectorData( [ '/cube' ] ) )
+		o["filter"].setInput( f["out"] )
+
+		# ensure the source position primitive variable interpolation is ver
+		self.assertEqual(c["out"].object("/cube")['P'].interpolation, IECore.PrimitiveVariable.Interpolation.Vertex )
+
+		o['in'].setInput( c["out"] )
+		o['interpolation'].setValue( IECore.PrimitiveVariable.Interpolation.Uniform )
+
+		inPoint = GafferOSL.OSLShader( "InPoint" )
+		s.addChild( inPoint )
+		inPoint.loadShader( "ObjectProcessing/InPoint", keepExistingValues = True )
+
+		vectorAdd = GafferOSL.OSLShader( "VectorAdd" )
+		s.addChild( vectorAdd )
+		vectorAdd.loadShader( "Maths/VectorAdd", keepExistingValues = True )
+		vectorAdd["parameters"]["b"].setValue( IECore.V3f( 1, 2, 3 ) )
+
+		vectorAdd["parameters"]["a"].setInput( inPoint["out"]["value"] )
+
+		outPoint = GafferOSL.OSLShader( "OutPoint" )
+		s.addChild( outPoint )
+		outPoint.loadShader( "ObjectProcessing/OutPoint", keepExistingValues=True )
+		outPoint['parameters']['name'].setValue("P_copy")
+
+		outPoint["parameters"]["value"].setInput( vectorAdd["out"]["out"] )
+
+		outObject = GafferOSL.OSLShader( "OutObject" )
+		s.addChild( outObject )
+		outObject.loadShader( "ObjectProcessing/OutObject", keepExistingValues=True )
+		outObject["parameters"]["in0"].setInput( outPoint["out"]["primitiveVariable"] )
+		outObject.loadShader( "ObjectProcessing/OutObject", keepExistingValues=True )
+
+		o["shader"].setInput( outObject["out"] )
+
+		cubeObject = s['OSLObject']['out'].object( "/cube" )
+
+		self.assertTrue( "P_copy" in cubeObject.keys() )
+		self.assertEqual( cubeObject["P_copy"].interpolation, IECore.PrimitiveVariable.Interpolation.Uniform)
+
+		self.assertEqual( cubeObject["P_copy"].data[0], IECore.V3f(  0.0,  0.0, -0.5 ) + IECore.V3f( 1, 2, 3 ))
+		self.assertEqual( cubeObject["P_copy"].data[1], IECore.V3f(  0.5,  0.0,  0.0 ) + IECore.V3f( 1, 2, 3 ))
+		self.assertEqual( cubeObject["P_copy"].data[2], IECore.V3f(  0.0,  0.0,  0.5 ) + IECore.V3f( 1, 2, 3 ))
+		self.assertEqual( cubeObject["P_copy"].data[3], IECore.V3f( -0.5,  0.0,  0.0 ) + IECore.V3f( 1, 2, 3 ))
+		self.assertEqual( cubeObject["P_copy"].data[4], IECore.V3f(  0.0,  0.5,  0.0 ) + IECore.V3f( 1, 2, 3 ))
+		self.assertEqual( cubeObject["P_copy"].data[5], IECore.V3f(  0.0, -0.5,  0.0 ) + IECore.V3f( 1, 2, 3 ))
+
+def testCanShadeFaceVaryingInterpolatedPrimitiveVariablesAsVertex( self ) :
```

Comment:
I should have squashed that intermediate step - sorry. I'm a little concerned sometimes I'm too eager to squash changes into fewer commits  but in this case it would have made things clearer.

I'm unclear if I need to update affects & hash to consider the resampledInPlug?